### PR TITLE
Task dev 3

### PR DESCRIPTION
**Pull request by NastassiaZhylinskaya**

### CODE REVIEW

File: DEV-3/DecimalNumberRadixConverter.cs [hunk 1]
```diff
@@ -0,0 +1,42 @@
+﻿using System;
+
+namespace NumberRadixConvertion
+{
+  /// <summary>
+  /// This class for converting decimal number
+  /// to another numeral system from 2 to 20.
+  /// </summary>
+  public class DecimalNumberRadixConvertor
+  {
```
Comment (IgorKheidorov) [inline DEV-3/DecimalNumberRadixConverter.cs]: Better LATIN_SYMBOLS_START_POSITION

File: DEV-3/DecimalNumberRadixConverter.cs [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  public class DecimalNumberRadixConvertor
+  {
+    const int ASCIISYMBOL = 55;
+    public int SourceNumber { get; private set; }
+        
+    public DecimalNumberRadixConvertor(int sourceNumber)
+    {
+      SourceNumber = sourceNumber;      
+    }
+    /// <summary>
```
Comment (IgorKheidorov) [inline DEV-3/DecimalNumberRadixConverter.cs]: method converts

File: DEV-3/DecimalNumberRadixConverter.cs [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    /// This method convert decimal number
```
Comment (IgorKheidorov) [inline DEV-3/DecimalNumberRadixConverter.cs]: to other

File: DEV-3/EntryPoint.cs [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+{
+  /// <summary>
+  /// It's an entry point in the program.
+  /// </summary>
+  class EntryPoint
+  {
+    static void Main(string[] args)
+    {
+      try
+      {
```
Comment (IgorKheidorov) [inline DEV-3/EntryPoint.cs]: Create local value for Parse(args[1]) not to call the method three times

File: DEV-3/DecimalNumberRadixConverter.cs [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+namespace NumberRadixConvertion
+{
+  /// <summary>
+  /// This class for converting decimal number
+  /// to another numeral system from 2 to 20.
+  /// </summary>
+  public class DecimalNumberRadixConvertor
+  {
+    const int ASCIISYMBOL = 55;
+    public int SourceNumber { get; private set; }
```
Comment (IgorKheidorov) [inline DEV-3/DecimalNumberRadixConverter.cs]: Why not use BigInteger?

**Review by IgorKheidorov — CHANGES_REQUESTED** [review]

File: DEV-3/DecimalNumberRadixConverter.cs [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+
+namespace NumberRadixConvertion
+{
+  /// <summary>
+  /// This class for converting decimal number
+  /// to other numeral system from 2 to 20.
+  /// </summary>
+  public class DecimalNumberRadixConvertor
+  {
+    const int LATIN_SYMBOLS_START_POSITION = 55;
```
Comment (AlexanderSoroka) [inline DEV-3/DecimalNumberRadixConverter.cs:12]: not obvious constant value

File: DEV-3/DecimalNumberRadixConverter.cs [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+    public BigInteger SourceNumber { get; private set; }
+        
+    public DecimalNumberRadixConvertor(BigInteger sourceNumber)
+    {
+      SourceNumber = sourceNumber;      
+    }
+    /// <summary>
```
Comment (AlexanderSoroka) [inline DEV-3/DecimalNumberRadixConverter.cs:19]: add blank line before new method description

File: DEV-3/EntryPoint.cs [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+namespace NumberRadixConvertion
+{
+  /// <summary>
+  /// It's an entry point in the program.
+  /// </summary>
+  class EntryPoint
+  {
+    static void Main(string[] args)
+    {
+      int basisOfANewNumeralSystem = int.Parse(args[1]);
```
Comment (AlexanderSoroka) [inline DEV-3/EntryPoint.cs:13]: Must be called within try/catch block

File: DEV-3/DecimalNumberRadixConverter.cs [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+      string convertedNumberString = String.Empty;
+      if (0 == sourceNumber)
+      {
+        return sourceNumber.ToString();
+      }
+      while (sourceNumber > 0)
+      {
+        int rest = sourceNumber % radix;
+        sourceNumber = sourceNumber / radix;
+        int asciSymbolFirstPosition = (rest >= 10) ? LATIN_SYMBOLS_START_POSITION : '0';
```
Comment (AlexanderSoroka) [inline DEV-3/DecimalNumberRadixConverter.cs:37]: Strange mix of magic number and constant

File: DEV-3/DecimalNumberRadixConverter.cs [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+    /// <summary>
+    /// This method converts decimal number
+    /// to other numeral system.
+    /// </summary>
+    /// <param name="radix">base numeral system.</param>
+    /// <returns>converted string.</returns>
+    public string ConvertTo(int radix)
+    {
+      BigInteger sourceNumber = SourceNumber;
+      string convertedNumberString = String.Empty;
```
Comment (AlexanderSoroka) [inline DEV-3/DecimalNumberRadixConverter.cs:28]: just 'result' would be much better

**Review by AlexanderSoroka — APPROVED** [review]
Approved due to deadline

**Pull request merged**